packages/remix-dev/cli/commands.ts

```diff
@@ -158,6 +180,12 @@ export async function dev(remixRoot: string, modeArg?: string) {
 
   let config = await readConfig(remixRoot);
   let mode = isBuildMode(modeArg) ? modeArg : BuildMode.Development;
+
+  // Load .env files if present.  Precedence: .env.local -> .env.{mode} -> .env
+  await loadEnv(config.rootDirectory, "local");
+  await loadEnv(config.rootDirectory, mode);
```

Comment:
`npm run dev` sets `NODE_ENV=development` but it does seem possible to use `NODE_ENV=production npx remix dev` so this currently handles both `.env.development` and `.env.production`.  If the intention is not to use `remix dev` with `production` then we can simplify this to only handle `.env.development`.

Reply:
The more I think about this, I'm not convinced all 3 are relevant for `remix dev` alone.  The original github issue was suggesting all 3 I think with the intention that they'd be used by adapters as well.  Since this is only for `remix dev`, I feel like only supporting `.env` will be sufficient?  I think if folks are doing advanced enough stuff to warrant `.production` and/or overrides in `.local`, it's not too much to leave that up to the user?

I can maybe see a use case for `.env.local` as an entry point for users to symlink to any number of specific local override files they care to maintain while not having to dup values from `.env` in them...

Reply:
I say let's start with only supporting `.env` and then we can add support for the different modes in the future if there's a need there.